Refactor rules to use lambdas

SamUpid opened the pull request:
Related to: #59"
Converted the following rules to use lambda functions as intermediate step:
- 1_1_1_2_5
- 7
- 10
- 11
- 13
- 16
- 19
- 20
- 21
- 23

This is preparation for named predicates (condition_<rule> or c_<rule>)

SamUpid commented:
@Bumblebee00 Now that the `_isone` issue is fixed.

Bumblebee00 commented:
yeah the speed up is not 17x, as rn in the package "solution r1" is implemented. Anyway work done so far is good. Now if you change in all this file all the one condition conditions into named predicates, if there are any remaining, i will merge it.

SamUpid commented:
Dear @Bumblebee00,

I have converted all single-variable conditions to named predicates.

Commit 1:  Converted rules 5, 19, 20, 21, 23, 32 (6 rules earlier)
Commit 2: Converted rules 7, 10, 11, 13, 18 (5 more rules).

SamUpid commented:
Ok, I will try to solve issue #59, and I will review and understand issue #77, and then let you know about it, so I can continue with your help and guidance.
